### PR TITLE
discover ci step

### PR DESCRIPTION
**Pull request by PietroPasotti**
This PR adds a 'discover' workflow action that allows us to run `interface_tester discover` on, for example, a PR branch.
The idea is that when reviewing e.g. #55 or #59 we can click on a 'Discover' button in the GH UI and see, for example:
```
Discovered:
login_ui_endpoints:
  - v0:
   - requirer:
     - <no tests>
     - schema NOT OK
     - <no charms>
   - provider:
     - <no tests>
     - schema OK
     - charms:
       - identity-platform-login-ui-operator (https://github.com/canonical/identity-platform-login-ui-operator) custom_test_setup=no
```

which tells us that the requirer schema is not correctly implemented and that no interface tests are present.
This makes reviewing PRs somewhat easier; otherwise it's harder to verify that tests being added or schemas being modified are correctly gathered by the same code that will eventually use them.

### CODE REVIEW

**Comment by PietroPasotti** [comment]
does our CI guru @lucabello know how to make the action configurable so that we can limit the scope to a specific interface (by name) and pass that to the discover command as an argument?
E.g. a dropdown with interface names --> `interface_tester discover --include login_ui_endpoints`

Also, how do we 'show' the results in a nice way?

**Review by gruyaume — APPROVED** [review]

**Comment by PietroPasotti** [comment]
> @PietroPasotti - this LGTM - is there anything outstanding here?

A quick look by @lucabello's keen automation eyes, then I'll merge

**Review by lucabello — COMMENTED** [review]
Looks good indeed; I proposed some changes to add the behavior you wanted of scoping the action call to specific interface names :)

File: .github/workflows/discover.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Discover interfaces
+
+on: workflow_dispatch
```
[the remark this answers is not in the document]
Comment (PietroPasotti) [reply .github/workflows/discover.yaml:3]: added default value. Thanks!

**Pull request merged**